src/model/RefrigerationSystem.cpp

```diff
@@ -189,26 +190,37 @@ namespace detail {
       result.insert(result.end(), removedTransferLoads.begin(), removedTransferLoads.end());
     }
 
+    // We're clearing the compressor/HighStage compressor list objects first. otherwise a compressor on the high stage list will call
+    // RefrgerationCompressor_Impl::system which will try to locate the compressorList which was deleted first
+    for (auto& c: compressors()) {
+      std::vector<IdfObject> removedCompressor = c.remove();
+      result.insert(result.end(), removedCompressor.begin(), removedCompressor.end());
+    }
+    for (auto& c: highStageCompressors()) {
+      std::vector<IdfObject> removedCompressor = c.remove();
+      result.insert(result.end(), removedCompressor.begin(), removedCompressor.end());
+    }
     if (boost::optional<ModelObjectList> compressorList = this->compressorList()) {
       std::vector<IdfObject> removedCompressors = compressorList->remove();
       result.insert(result.end(), removedCompressors.begin(), removedCompressors.end());
     }
+    if (boost::optional<ModelObjectList> highStageCompressorList = this->highStageCompressorList()) {
+      std::vector<IdfObject> removedHighStageCompressors = highStageCompressorList->remove();
+      result.insert(result.end(), removedHighStageCompressors.begin(), removedHighStageCompressors.end());
+    }
 
+    // TODO: technically this object can be shared accross multiple systems..
     if ( boost::optional<RefrigerationSubcoolerMechanical> mechSubcooler = this->mechanicalSubcooler() ) {
       std::vector<IdfObject> removedMechSubcooler = mechSubcooler->remove();
       result.insert(result.end(), removedMechSubcooler.begin(), removedMechSubcooler.end());
     }
 
+    // TODO: technically this object can be shared accross multiple systems..
     if ( boost::optional<RefrigerationSubcoolerLiquidSuction> liqSuctionSubcooler = this->liquidSuctionHeatExchangerSubcooler() ) {
```

Comment:
Should be disallow sharing `RefrigerationSubcoolerLiquidSuction` accross several RefrigerationSystem? Or not delete it unless it becomes orphaned? @kbenne ?